### PR TITLE
Adding Dapr Container default wait strategy based on healthz/outbound

### PR DESCRIPTION
**Pull request by artur-ciocanu**
# Description

Adding Dapr Container default wait strategy based on healthz/outbound

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #1104

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [ ] Extended the documentation

### CODE REVIEW

**Comment by artur-ciocanu** [comment]
@artursouza and @cicoyle could you please take a look. This is required to make sure that we declare Dapr Testcontainer ready only when all the components are initialized.

Thank you!

File: sdk-tests/pom.xml [hunk 1]
```diff
@@ -25,6 +25,8 @@
     <springboot.version>3.3.1</springboot.version>
     <logback-classic.version>1.4.12</logback-classic.version>
     <wiremock.version>3.9.1</wiremock.version>
+    <testcontainers-dapr.version>0.13.0-SNAPSHOT</testcontainers-dapr.version>
```
Comment (artursouza) [inline sdk-tests/pom.xml:28]: Change the version update script in .github/scripts to also update this on release.

**Review by artursouza — COMMENTED** [review]
Just a minor comment. LGTM.

**Comment by artur-ciocanu** [comment]
> Just a minor comment. LGTM. 

Fixed.

**Comment by artur-ciocanu** [comment]
@artursouza and @cicoyle the PR has been adjusted. The `Validate Examples` failed because of some Docker Compose diff in output. I haven't touched any of that. It would be good to see if we can re-trigger the build, to ensure things look good and merge the PR.

**Review by cicoyle — APPROVED** [review]
lgtm

**Review by artursouza — APPROVED** [review]

**Pull request merged**